### ISSUE TITLE
pdb debugger command <ll> (longlist shows wrong current line in python versions 3.10, 3.11

### ISSUE THREAD

**Issue by Th1sUs3rH4sN0N4m3** [python/cpython#104207]
In python version 3.10 and 3.11 pdb debugger has a bug. If you run `python3.11 -m pdb ./your_file_name.py`, you will see next
```bash

(venv) user@laptop% python3.11 -m pdb ./main.py
> Path/to/file/main.py(1)<module>()
-> a = 10
(Pdb) 

```
if you type command `l` (list) you wiil get 
```bash
(Pdb) l
  1  -> a = 10
  2     b = a + 5
  3     c = a + b
  4     
  5     print(c)
[EOF]
(Pdb) 
```
But if you type `ll` (long list), you will see next
```bash
(Pdb) ll
  0     a = 10
  1  -> b = a + 5
  2     c = a + b
  3     
  4     print(c)
(Pdb) 

```
you can see cursor  '->' show different positions.
I reproduced it for version 3.10 and 3.11
In python version 3.8 everything is fine. There is no problem. The position is the same.

# My environment
Macos Monteray 12.6.5  CPU Intel

python 3.11
python 3.10
python 3.8

**Comment by gaogaotiantian** [comment]
I believe this is fixed by https://github.com/python/cpython/pull/103265 . 3.10 stopped taking bug fixes now, I don't know why it was not ported back in 3.11. I'll label this as pending for now.

**Comment by gaogaotiantian** [comment]
The fix is merged back in 3.11 https://github.com/python/cpython/pull/104262. Close as fixed.
